kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/DeleteCommand.java

```diff
@@ -46,11 +51,24 @@ public int run() throws IOException {
         View view = Datasets.load(uriOrName);
         Preconditions.checkArgument(viewMatches(view.getUri(), uriOrName),
             "Resolved view does not match requested view: " + view.getUri());
-        view.deleteAll();
+
+        if(console.isDebugEnabled()){
+          for(PartitionView<?> pv: (Iterable<PartitionView<?>>) view.getCoveringPartitions()){
+            console.debug("Deleting partition {}", pv.getLocation());
+          }
+        }
+
+        if(!dryRun) {
+          view.deleteAll();
+        }
       } else if (isDatasetUri(uriOrName)) {
-        Datasets.delete(uriOrName);
+        if (!dryRun){
+          Datasets.delete(uriOrName);
+        }
       } else {
-        getDatasetRepository().delete(namespace, uriOrName);
+        if(!dryRun) {
+          getDatasetRepository().delete(namespace, uriOrName);
+        }
       }
       console.debug("Deleted {}", uriOrName);
```

Comment:
I think this should change depending on whether this is a dry run. If so, it should say "Would have deleted {}" or something similar to signal that no action was taken.